velox/functions/lib/Re2Functions.cpp

```diff
@@ -1111,31 +1113,32 @@ class PatternStringIterator {
   };
 
   // Char at current cursor.
-  char current() const {
-    return pattern_.data()[currentIndex_];
+  char charAt(size_t index) const {
+    VELOX_DCHECK(index >= 0 && index < pattern_.size())
+    return pattern_.data()[index];
   }
 
   const StringView pattern_;
   const std::optional<char> escapeChar_;
-  const size_t lastIndex_;
 
-  int32_t currentIndex_{-1};
+  size_t currentStart_{0};
+  size_t nextStart_{0};
   CharKind charKind_{CharKind::kNormal};
   bool isPreviousWildcard_{false};
 };
 
 PatternMetadata determinePatternKind(
     StringView pattern,
     std::optional<char> escapeChar) {
-  int32_t patternLength = pattern.size();
+  const size_t patternLength = pattern.size();
 
   // Index of the first % or _ character(not escaped).
-  int32_t wildcardStart = -1;
+  size_t wildcardStart = -1;
```

Comment:
wildcardStart is a index, if it is not found, it could be negative(-1), while `size_t` is unsigned?

Reply:
Yes. I'll change it back to `int32_t`. Just missed that the values are `-1`s.